atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,265 @@
+
+        if not tag_info or 'name' not in tag_info:
+            self.log.warning("No tag info was retrieved")
+            return
+
+        repo_info = xmlrpc.getRepo(tag_info['id'])
+
+        if not repo_info or 'id' not in repo_info:
+            self.log.warning("No repo info was retrieved")
+            return
+
+        # to use urljoin, we would have to append '/', so let's append everything
+        baseurl = pathinfo.repo(repo_info['id'], tag_info['name']) + "/$basearch"
+
+        self.log.info("baseurl = '%s'", baseurl)
+
+        repo = {
+            'name': 'atomic-reactor-koji-plugin-%s' % self.target,
+            'baseurl': baseurl,
+            'enabled': 1,
+            'gpgcheck': 0,
+        }
+
+        # yum doesn't accept a certificate path in sslcacert - it requires a db with added cert
+        # dnf ignores that option completely
+        # we have to fall back to sslverify=0 everytime we get https repo from brew so we'll surely
+        # be able to pull from it
+
+        if baseurl.startswith("https://"):
+            self.log.info("Ignoring certificates in the repo")
+            repo['sslverify'] = 0
+
+        if proxy:
+            self.log.info("Setting yum proxy to %s", proxy)
+            repo['proxy'] = proxy
+
+        yum_repo = YumRepo(os.path.join(YUM_REPOS_DIR, self.target))
+        path = yum_repo.dst_filename
+        self.log.info("yum repo of koji target: '%s'", path)
+        yum_repo.content = render_yum_repo(repo, escape_dollars=False)
+        for platform in self.platforms:
+            self.yum_repos[platform].append(yum_repo)
+
+    def _inject_into_repo_files(self, build_dir: BuildDir):
+        """Inject repo files into a relative directory inside the build context"""
+        host_repos_path = build_dir.path / RELATIVE_REPOS_PATH
+        self.log.info("creating directory for yum repos: %s", host_repos_path)
+        os.mkdir(host_repos_path)
+        allow_repo_dir_in_dockerignore(build_dir.path)
+
+        for repo in self.yum_repos[build_dir.platform]:
+            # Update every repo accordingly in a repofile
+            # input_buf ---- updated ----> updated_buf
+            with StringIO(repo.content.decode()) as input_buf, StringIO() as updated_buf:
+                for line in input_buf:
+                    updated_buf.write(line)
+                    # Apply sslcacert to every repo in a repofile
+                    if line.lstrip().startswith('[') and self._builder_ca_bundle:
+                        updated_buf.write(f'sslcacert=/tmp/{self._ca_bundle_pem}\n')
+
+                yum_repo = YumRepo(repourl=repo.dst_filename,
+                                   content=updated_buf.getvalue(),
+                                   dst_repos_dir=host_repos_path,
+                                   add_hash=False)
+                yum_repo.write_content()
+
+    def _inject_into_dockerfile(self, build_dir: BuildDir):
+        self._dockerfile = build_dir.dockerfile
```

Comment:
Let's not initialize it here. And it should be `dockerfile_with_parent_env()`, the USER line may reference a parent variable

Reply:
changed to use local var and use `dockerfile_with_parent_env()`